tests/testthat/test-scatter_plot_financial.R

```diff
@@ -21,23 +21,16 @@ test_that("returns correct benchmarks values", {
 
 test_that("calculate_rank handles NAs for any mode", {
   data <- tibble(
+    bank_id = c("a", "b"),
     profile_ranking = c(1, NA),
     equal_weight_finance = c(1, 1),
     best_case_finance = c(1, 1),
     worst_case_finance = c(1, 1)
   )
-  results <- c(
-    calculate_rank(data, "equal_weight_finance", "profile_ranking"),
-    calculate_rank(data, "best_case_finance", "profile_ranking"),
-    calculate_rank(data, "worst_case_finance", "profile_ranking")
+  results <- list(
+    calculate_rank(data, "equal_weight_finance", aka("profile_ranking")),
+    calculate_rank(data, "best_case_finance", aka("profile_ranking")),
+    calculate_rank(data, "worst_case_finance", aka("profile_ranking"))
   )
-  expect_true(all(results == 1))
-})
-
-test_that("prepare_scatter_plot_financial removes NAs", {
-  test_fin <- financial
-  test_fin[3, "reduction_targets"] <- NA
-  result <- prepare_scatter_plot_financial(test_fin, "all", "IPR", 2030)
-
-  expect_true(all(!is.na(result$percent)))
+  expect_true(all(sapply(results, function(x) all(x$rank == 1))))
```

Comment:
In general it's best to avoid `sapply()`. You may use `vapply()`.

> Base R has two apply functions that can return atomic vectors: [sapply()](https://rdrr.io/r/base/lapply.html) and [vapply()](https://rdrr.io/r/base/lapply.html). I recommend that you avoid [sapply()](https://rdrr.io/r/base/lapply.html) because it tries to simplify the result, so it can return a list, a vector, or a matrix. This makes it difficult to program with, and it should be avoided in non-interactive settings.

https://adv-r.hadley.nz/functionals.html?q=sapply#map-atomic


I rarely use `vapply()` (I find it verbose). Instead I typically use the pattern `lapply(...) |> unlist()` -- which is two functions instead of one but they compose naturally in a clear way.